// @irsdk-node/types/dist/esm/index.js
// src/defines.ts
var UNLIMITED_LAPS_COUNT = 32767;
var UNLIMITED_TIME_VALUE = 604800;
var StatusField = /* @__PURE__ */ ((StatusField2) => {
  StatusField2[StatusField2["Connected"] = 1] = "Connected";
  return StatusField2;
})(StatusField || {});
var VarTypes = /* @__PURE__ */ ((VarTypes2) => {
  VarTypes2[VarTypes2["Character"] = 0] = "Character";
  VarTypes2[VarTypes2["Boolean"] = 1] = "Boolean";
  VarTypes2[VarTypes2["Integer"] = 2] = "Integer";
  VarTypes2[VarTypes2["BitField"] = 3] = "BitField";
  VarTypes2[VarTypes2["Float"] = 4] = "Float";
  VarTypes2[VarTypes2["Double"] = 5] = "Double";
  VarTypes2[VarTypes2["ETCount"] = 6] = "ETCount";
  return VarTypes2;
})(VarTypes || {});
var VarTypesReadable = {
  [0 /* Character */]: "string",
  [1 /* Boolean */]: "boolean",
  [2 /* Integer */]: "number",
  [3 /* BitField */]: "number",
  [4 /* Float */]: "number",
  [5 /* Double */]: "number",
  [6 /* ETCount */]: "never"
};
var VarTypeBytes = {
  [0 /* Character */]: 1,
  [1 /* Boolean */]: 1,
  [2 /* Integer */]: 4,
  [3 /* BitField */]: 4,
  [4 /* Float */]: 4,
  [5 /* Double */]: 4,
  [6 /* ETCount */]: 6 /* ETCount */
};
var TrackLocation = /* @__PURE__ */ ((TrackLocation2) => {
  TrackLocation2[TrackLocation2["NotInWorld"] = -1] = "NotInWorld";
  TrackLocation2[TrackLocation2["OffTrack"] = 0] = "OffTrack";
  TrackLocation2[TrackLocation2["InPitStall"] = 1] = "InPitStall";
  TrackLocation2[TrackLocation2["ApproachingPits"] = 2] = "ApproachingPits";
  TrackLocation2[TrackLocation2["OnTrack"] = 3] = "OnTrack";
  return TrackLocation2;
})(TrackLocation || {});
var TrackSurface = /* @__PURE__ */ ((TrackSurface2) => {
  TrackSurface2[TrackSurface2["SurfaceNotInWorld"] = -1] = "SurfaceNotInWorld";
  TrackSurface2[TrackSurface2["UndefinedMaterial"] = 0] = "UndefinedMaterial";
  TrackSurface2[TrackSurface2["Asphalt1Material"] = 1] = "Asphalt1Material";
  TrackSurface2[TrackSurface2["Asphalt2Material"] = 2] = "Asphalt2Material";
  TrackSurface2[TrackSurface2["Asphalt3Material"] = 3] = "Asphalt3Material";
  TrackSurface2[TrackSurface2["Asphalt4Material"] = 4] = "Asphalt4Material";
  TrackSurface2[TrackSurface2["Concrete1Material"] = 5] = "Concrete1Material";
  TrackSurface2[TrackSurface2["Concrete2Material"] = 6] = "Concrete2Material";
  TrackSurface2[TrackSurface2["RacingDirt1Material"] = 7] = "RacingDirt1Material";
  TrackSurface2[TrackSurface2["RacingDirt2Material"] = 8] = "RacingDirt2Material";
  TrackSurface2[TrackSurface2["Paint1Material"] = 9] = "Paint1Material";
  TrackSurface2[TrackSurface2["Paint2Material"] = 10] = "Paint2Material";
  TrackSurface2[TrackSurface2["Rumble1Material"] = 11] = "Rumble1Material";
  TrackSurface2[TrackSurface2["Rumble2Material"] = 12] = "Rumble2Material";
  TrackSurface2[TrackSurface2["Rumble3Material"] = 13] = "Rumble3Material";
  TrackSurface2[TrackSurface2["Rumble4Material"] = 14] = "Rumble4Material";
  TrackSurface2[TrackSurface2["Grass1Material"] = 15] = "Grass1Material";
  TrackSurface2[TrackSurface2["Grass2Material"] = 16] = "Grass2Material";
  TrackSurface2[TrackSurface2["Grass3Material"] = 17] = "Grass3Material";
  TrackSurface2[TrackSurface2["Grass4Material"] = 18] = "Grass4Material";
  TrackSurface2[TrackSurface2["Dirt1Material"] = 19] = "Dirt1Material";
  TrackSurface2[TrackSurface2["Dirt2Material"] = 20] = "Dirt2Material";
  TrackSurface2[TrackSurface2["Dirt3Material"] = 21] = "Dirt3Material";
  TrackSurface2[TrackSurface2["Dirt4Material"] = 22] = "Dirt4Material";
  TrackSurface2[TrackSurface2["SandMaterial"] = 23] = "SandMaterial";
  TrackSurface2[TrackSurface2["Gravel1Material"] = 24] = "Gravel1Material";
  TrackSurface2[TrackSurface2["Gravel2Material"] = 25] = "Gravel2Material";
  TrackSurface2[TrackSurface2["GrasscreteMaterial"] = 26] = "GrasscreteMaterial";
  TrackSurface2[TrackSurface2["AstroturfMaterial"] = 27] = "AstroturfMaterial";
  return TrackSurface2;
})(TrackSurface || {});
var SessionState = /* @__PURE__ */ ((SessionState2) => {
  SessionState2[SessionState2["Invalid"] = 0] = "Invalid";
  SessionState2[SessionState2["GetInCar"] = 1] = "GetInCar";
  SessionState2[SessionState2["Warmup"] = 2] = "Warmup";
  SessionState2[SessionState2["ParadeLaps"] = 3] = "ParadeLaps";
  SessionState2[SessionState2["Racing"] = 4] = "Racing";
  SessionState2[SessionState2["Checkered"] = 5] = "Checkered";
  SessionState2[SessionState2["CoolDown"] = 6] = "CoolDown";
  return SessionState2;
})(SessionState || {});
var CarLeftRight = /* @__PURE__ */ ((_CarLeftRight) => {
  _CarLeftRight[_CarLeftRight["Off"] = 0] = "Off";
  _CarLeftRight[_CarLeftRight["Clear"] = 1] = "Clear";
  _CarLeftRight[_CarLeftRight["CarLeft"] = 2] = "CarLeft";
  _CarLeftRight[_CarLeftRight["CarRight"] = 3] = "CarRight";
  _CarLeftRight[_CarLeftRight["CarLeftRight"] = 4] = "CarLeftRight";
  _CarLeftRight[_CarLeftRight["Cars2Left"] = 5] = "Cars2Left";
  _CarLeftRight[_CarLeftRight["Cars2Right"] = 6] = "Cars2Right";
  return _CarLeftRight;
})(CarLeftRight || {});
var PitSvStatus = /* @__PURE__ */ ((PitSvStatus2) => {
  PitSvStatus2[PitSvStatus2["None"] = 0] = "None";
  PitSvStatus2[PitSvStatus2["InProgress"] = 1] = "InProgress";
  PitSvStatus2[PitSvStatus2["Complete"] = 2] = "Complete";
  PitSvStatus2[PitSvStatus2["TooFarLeft"] = 100] = "TooFarLeft";
  PitSvStatus2[PitSvStatus2["TooFarRight"] = 101] = "TooFarRight";
  PitSvStatus2[PitSvStatus2["TooFarForward"] = 102] = "TooFarForward";
  PitSvStatus2[PitSvStatus2["TooFarBack"] = 103] = "TooFarBack";
  PitSvStatus2[PitSvStatus2["BadAngle"] = 104] = "BadAngle";
  PitSvStatus2[PitSvStatus2["CantFixThat"] = 105] = "CantFixThat";
  return PitSvStatus2;
})(PitSvStatus || {});
var PaceMode = /* @__PURE__ */ ((PaceMode2) => {
  PaceMode2[PaceMode2["SingleFileStart"] = 0] = "SingleFileStart";
  PaceMode2[PaceMode2["DoubleFileStart"] = 1] = "DoubleFileStart";
  PaceMode2[PaceMode2["SingleFileRestart"] = 2] = "SingleFileRestart";
  PaceMode2[PaceMode2["DoubleFileRestart"] = 3] = "DoubleFileRestart";
  PaceMode2[PaceMode2["NotPacing"] = 4] = "NotPacing";
  return PaceMode2;
})(PaceMode || {});
var TrackWetness = /* @__PURE__ */ ((TrackWetness2) => {
  TrackWetness2[TrackWetness2["UNKNOWN"] = 0] = "UNKNOWN";
  TrackWetness2[TrackWetness2["Dry"] = 1] = "Dry";
  TrackWetness2[TrackWetness2["MostlyDry"] = 2] = "MostlyDry";
  TrackWetness2[TrackWetness2["VeryLightlyWet"] = 3] = "VeryLightlyWet";
  TrackWetness2[TrackWetness2["LightlyWet"] = 4] = "LightlyWet";
  TrackWetness2[TrackWetness2["ModeratelyWet"] = 5] = "ModeratelyWet";
  TrackWetness2[TrackWetness2["VeryWet"] = 6] = "VeryWet";
  TrackWetness2[TrackWetness2["ExtremelyWet"] = 7] = "ExtremelyWet";
  return TrackWetness2;
})(TrackWetness || {});
var IncidentFlags = /* @__PURE__ */ ((IncidentFlags2) => {
  IncidentFlags2[IncidentFlags2["NoReport"] = 0] = "NoReport";
  IncidentFlags2[IncidentFlags2["ReportOutOfControl"] = 1] = "ReportOutOfControl";
  IncidentFlags2[IncidentFlags2["ReportOffTrack"] = 2] = "ReportOffTrack";
  IncidentFlags2[IncidentFlags2["ReportOffTrackOngoing"] = 3] = "ReportOffTrackOngoing";
  IncidentFlags2[IncidentFlags2["ReportContactWithWorld"] = 4] = "ReportContactWithWorld";
  IncidentFlags2[IncidentFlags2["ReportCollisionWithWorld"] = 5] = "ReportCollisionWithWorld";
  IncidentFlags2[IncidentFlags2["ReportCollisionWithWorldOngoing"] = 6] = "ReportCollisionWithWorldOngoing";
  IncidentFlags2[IncidentFlags2["ReportContactWithCar"] = 7] = "ReportContactWithCar";
  IncidentFlags2[IncidentFlags2["ReportCollisionWithCar"] = 8] = "ReportCollisionWithCar";
  IncidentFlags2[IncidentFlags2["PenaltyZeroX"] = 256] = "PenaltyZeroX";
  IncidentFlags2[IncidentFlags2["PenaltyOneX"] = 512] = "PenaltyOneX";
  IncidentFlags2[IncidentFlags2["PenaltyTwoX"] = 768] = "PenaltyTwoX";
  IncidentFlags2[IncidentFlags2["PenaltyFourX"] = 1024] = "PenaltyFourX";
  return IncidentFlags2;
})(IncidentFlags || {});
var IncidentFlagMask = /* @__PURE__ */ ((IncidentFlagMask2) => {
  IncidentFlagMask2[IncidentFlagMask2["Report"] = 255] = "Report";
  IncidentFlagMask2[IncidentFlagMask2["Penalty"] = 65280] = "Penalty";
  return IncidentFlagMask2;
})(IncidentFlagMask || {});
var EngineWarnings = /* @__PURE__ */ ((EngineWarnings2) => {
  EngineWarnings2[EngineWarnings2["WaterTempWarning"] = 1] = "WaterTempWarning";
  EngineWarnings2[EngineWarnings2["FuelPressureWarning"] = 2] = "FuelPressureWarning";
  EngineWarnings2[EngineWarnings2["OilPressureWarning"] = 4] = "OilPressureWarning";
  EngineWarnings2[EngineWarnings2["EngineStalled"] = 8] = "EngineStalled";
  EngineWarnings2[EngineWarnings2["PitSpeedLimiter"] = 16] = "PitSpeedLimiter";
  EngineWarnings2[EngineWarnings2["RevLimiterActive"] = 32] = "RevLimiterActive";
  EngineWarnings2[EngineWarnings2["OilTempWarning"] = 64] = "OilTempWarning";
  EngineWarnings2[EngineWarnings2["MandatoryRepairsNeeded"] = 128] = "MandatoryRepairsNeeded";
  EngineWarnings2[EngineWarnings2["OptionalRepairsNeeded"] = 256] = "OptionalRepairsNeeded";
  return EngineWarnings2;
})(EngineWarnings || {});
var GlobalFlags = /* @__PURE__ */ ((GlobalFlags2) => {
  GlobalFlags2[GlobalFlags2["Checkered"] = 1] = "Checkered";
  GlobalFlags2[GlobalFlags2["White"] = 2] = "White";
  GlobalFlags2[GlobalFlags2["Green"] = 4] = "Green";
  GlobalFlags2[GlobalFlags2["Yellow"] = 8] = "Yellow";
  GlobalFlags2[GlobalFlags2["Red"] = 16] = "Red";
  GlobalFlags2[GlobalFlags2["Blue"] = 32] = "Blue";
  GlobalFlags2[GlobalFlags2["Debris"] = 64] = "Debris";
  GlobalFlags2[GlobalFlags2["Crossed"] = 128] = "Crossed";
  GlobalFlags2[GlobalFlags2["YellowWaving"] = 256] = "YellowWaving";
  GlobalFlags2[GlobalFlags2["OneLapToGreen"] = 512] = "OneLapToGreen";
  GlobalFlags2[GlobalFlags2["GreenHeld"] = 1024] = "GreenHeld";
  GlobalFlags2[GlobalFlags2["TenToGo"] = 2048] = "TenToGo";
  GlobalFlags2[GlobalFlags2["FiveToGo"] = 4096] = "FiveToGo";
  GlobalFlags2[GlobalFlags2["RandomWaving"] = 8192] = "RandomWaving";
  GlobalFlags2[GlobalFlags2["Caution"] = 16384] = "Caution";
  GlobalFlags2[GlobalFlags2["CautionWaving"] = 32768] = "CautionWaving";
  GlobalFlags2[GlobalFlags2["Black"] = 65536] = "Black";
  GlobalFlags2[GlobalFlags2["Disqualify"] = 131072] = "Disqualify";
  GlobalFlags2[GlobalFlags2["Servicible"] = 262144] = "Servicible";
  GlobalFlags2[GlobalFlags2["Furled"] = 524288] = "Furled";
  GlobalFlags2[GlobalFlags2["Repair"] = 1048576] = "Repair";
  GlobalFlags2[GlobalFlags2["DqScoringInvalid"] = 2097152] = "DqScoringInvalid";
  GlobalFlags2[GlobalFlags2["StartHidden"] = 268435456] = "StartHidden";
  GlobalFlags2[GlobalFlags2["StartReady"] = 536870912] = "StartReady";
  GlobalFlags2[GlobalFlags2["StartSet"] = 1073741824] = "StartSet";
  GlobalFlags2[GlobalFlags2["StartGo"] = 2147483648] = "StartGo";
  return GlobalFlags2;
})(GlobalFlags || {});
var CameraState = /* @__PURE__ */ ((CameraState2) => {
  CameraState2[CameraState2["IsSessionScreen"] = 1] = "IsSessionScreen";
  CameraState2[CameraState2["IsScenicActive"] = 2] = "IsScenicActive";
  CameraState2[CameraState2["CamToolActive"] = 4] = "CamToolActive";
  CameraState2[CameraState2["UIHidden"] = 8] = "UIHidden";
  CameraState2[CameraState2["UseAutoShotSelection"] = 16] = "UseAutoShotSelection";
  CameraState2[CameraState2["UseTemporaryEdits"] = 32] = "UseTemporaryEdits";
  CameraState2[CameraState2["UseKeyAcceleration"] = 64] = "UseKeyAcceleration";
  CameraState2[CameraState2["UseKey10xAcceleration"] = 128] = "UseKey10xAcceleration";
  CameraState2[CameraState2["UseMouseAimMode"] = 256] = "UseMouseAimMode";
  return CameraState2;
})(CameraState || {});
var PitSvFlags = /* @__PURE__ */ ((PitSvFlags2) => {
  PitSvFlags2[PitSvFlags2["LFTireChange"] = 1] = "LFTireChange";
  PitSvFlags2[PitSvFlags2["RFTireChange"] = 2] = "RFTireChange";
  PitSvFlags2[PitSvFlags2["LRTireChange"] = 4] = "LRTireChange";
  PitSvFlags2[PitSvFlags2["RRTireChange"] = 8] = "RRTireChange";
  PitSvFlags2[PitSvFlags2["FuelFill"] = 16] = "FuelFill";
  PitSvFlags2[PitSvFlags2["WindshieldTearoff"] = 32] = "WindshieldTearoff";
  PitSvFlags2[PitSvFlags2["FastRepair"] = 64] = "FastRepair";
  return PitSvFlags2;
})(PitSvFlags || {});
var PaceFlags = /* @__PURE__ */ ((PaceFlags2) => {
  PaceFlags2[PaceFlags2["EndOfLine"] = 1] = "EndOfLine";
  PaceFlags2[PaceFlags2["FreePass"] = 2] = "FreePass";
  PaceFlags2[PaceFlags2["WavedAround"] = 4] = "WavedAround";
  return PaceFlags2;
})(PaceFlags || {});
var BroadcastMessages = /* @__PURE__ */ ((BroadcastMessages2) => {
  BroadcastMessages2[BroadcastMessages2["CameraSwitchPos"] = 0] = "CameraSwitchPos";
  BroadcastMessages2[BroadcastMessages2["CameraSwitchNum"] = 1] = "CameraSwitchNum";
  BroadcastMessages2[BroadcastMessages2["CameraSetState"] = 2] = "CameraSetState";
  BroadcastMessages2[BroadcastMessages2["ReplaySetPlaySpeed"] = 3] = "ReplaySetPlaySpeed";
  BroadcastMessages2[BroadcastMessages2["ReplaySetPlayPosition"] = 4] = "ReplaySetPlayPosition";
  BroadcastMessages2[BroadcastMessages2["ReplaySearch"] = 5] = "ReplaySearch";
  BroadcastMessages2[BroadcastMessages2["ReplaySetState"] = 6] = "ReplaySetState";
  BroadcastMessages2[BroadcastMessages2["ReloadTextures"] = 7] = "ReloadTextures";
  BroadcastMessages2[BroadcastMessages2["ChatCommand"] = 8] = "ChatCommand";
  BroadcastMessages2[BroadcastMessages2["PitCommand"] = 9] = "PitCommand";
  BroadcastMessages2[BroadcastMessages2["TelemCommand"] = 10] = "TelemCommand";
  BroadcastMessages2[BroadcastMessages2["FFBCommand"] = 11] = "FFBCommand";
  BroadcastMessages2[BroadcastMessages2["ReplaySearchSessionTime"] = 12] = "ReplaySearchSessionTime";
  BroadcastMessages2[BroadcastMessages2["VideoCapture"] = 13] = "VideoCapture";
  BroadcastMessages2[BroadcastMessages2["UnusedPlaceholder"] = 14] = "UnusedPlaceholder";
  return BroadcastMessages2;
})(BroadcastMessages || {});
var BroadcastMessageNames = {
  [0 /* CameraSwitchPos */]: "CameraSwitchPos",
  [1 /* CameraSwitchNum */]: "CameraSwitchNum",
  [2 /* CameraSetState */]: "CameraSetState",
  [3 /* ReplaySetPlaySpeed */]: "ReplaySetPlaySpeed",
  [4 /* ReplaySetPlayPosition */]: "ReplaySetPlayPosition",
  [5 /* ReplaySearch */]: "ReplaySearch",
  [6 /* ReplaySetState */]: "ReplaySetState",
  [7 /* ReloadTextures */]: "ReloadTextures",
  [8 /* ChatCommand */]: "ChatCommand",
  [9 /* PitCommand */]: "PitCommand",
  [10 /* TelemCommand */]: "TelemCommand",
  [11 /* FFBCommand */]: "FFBCommand",
  [12 /* ReplaySearchSessionTime */]: "ReplaySearchSessionTime",
  [13 /* VideoCapture */]: "VideoCapture",
  [14 /* UnusedPlaceholder */]: "UnusedPlaceholder"
};
var ChatCommand = /* @__PURE__ */ ((ChatCommand2) => {
  ChatCommand2[ChatCommand2["Macro"] = 0] = "Macro";
  ChatCommand2[ChatCommand2["BeginChat"] = 1] = "BeginChat";
  ChatCommand2[ChatCommand2["Reply"] = 2] = "Reply";
  ChatCommand2[ChatCommand2["Cancel"] = 3] = "Cancel";
  return ChatCommand2;
})(ChatCommand || {});
var PitCommand = /* @__PURE__ */ ((PitCommand2) => {
  PitCommand2[PitCommand2["Clear"] = 0] = "Clear";
  PitCommand2[PitCommand2["WS"] = 1] = "WS";
  PitCommand2[PitCommand2["Fuel"] = 2] = "Fuel";
  PitCommand2[PitCommand2["LF"] = 3] = "LF";
  PitCommand2[PitCommand2["RF"] = 4] = "RF";
  PitCommand2[PitCommand2["LR"] = 5] = "LR";
  PitCommand2[PitCommand2["RR"] = 6] = "RR";
  PitCommand2[PitCommand2["ClearTires"] = 7] = "ClearTires";
  PitCommand2[PitCommand2["FR"] = 8] = "FR";
  PitCommand2[PitCommand2["ClearWS"] = 9] = "ClearWS";
  PitCommand2[PitCommand2["ClearFR"] = 10] = "ClearFR";
  PitCommand2[PitCommand2["ClearFuel"] = 11] = "ClearFuel";
  PitCommand2[PitCommand2["ChangeTireCompound"] = 12] = "ChangeTireCompound";
  return PitCommand2;
})(PitCommand || {});
var TelemetryCommand = /* @__PURE__ */ ((TelemetryCommand2) => {
  TelemetryCommand2[TelemetryCommand2["Stop"] = 0] = "Stop";
  TelemetryCommand2[TelemetryCommand2["Start"] = 1] = "Start";
  TelemetryCommand2[TelemetryCommand2["Restart"] = 2] = "Restart";
  return TelemetryCommand2;
})(TelemetryCommand || {});
var ReplayStateCommand = /* @__PURE__ */ ((ReplayStateCommand2) => {
  ReplayStateCommand2[ReplayStateCommand2["EraseTape"] = 0] = "EraseTape";
  ReplayStateCommand2[ReplayStateCommand2["Last"] = 1] = "Last";
  return ReplayStateCommand2;
})(ReplayStateCommand || {});
var ReloadTexturesCommand = /* @__PURE__ */ ((ReloadTexturesCommand2) => {
  ReloadTexturesCommand2[ReloadTexturesCommand2["All"] = 0] = "All";
  ReloadTexturesCommand2[ReloadTexturesCommand2["CarIndex"] = 1] = "CarIndex";
  return ReloadTexturesCommand2;
})(ReloadTexturesCommand || {});
var ReplaySearchCommand = /* @__PURE__ */ ((ReplaySearchCommand2) => {
  ReplaySearchCommand2[ReplaySearchCommand2["ToStart"] = 0] = "ToStart";
  ReplaySearchCommand2[ReplaySearchCommand2["ToEnd"] = 1] = "ToEnd";
  ReplaySearchCommand2[ReplaySearchCommand2["PrevSession"] = 2] = "PrevSession";
  ReplaySearchCommand2[ReplaySearchCommand2["NextSession"] = 3] = "NextSession";
  ReplaySearchCommand2[ReplaySearchCommand2["PrevLap"] = 4] = "PrevLap";
  ReplaySearchCommand2[ReplaySearchCommand2["NextLap"] = 5] = "NextLap";
  ReplaySearchCommand2[ReplaySearchCommand2["PrevFrame"] = 6] = "PrevFrame";
  ReplaySearchCommand2[ReplaySearchCommand2["NextFrame"] = 7] = "NextFrame";
  ReplaySearchCommand2[ReplaySearchCommand2["PrevIncident"] = 8] = "PrevIncident";
  ReplaySearchCommand2[ReplaySearchCommand2["NextIncident"] = 9] = "NextIncident";
  ReplaySearchCommand2[ReplaySearchCommand2["UnusedPlaceholder"] = 10] = "UnusedPlaceholder";
  return ReplaySearchCommand2;
})(ReplaySearchCommand || {});
var ReplayPositionCommand = /* @__PURE__ */ ((ReplayPositionCommand2) => {
  ReplayPositionCommand2[ReplayPositionCommand2["Begin"] = 0] = "Begin";
  ReplayPositionCommand2[ReplayPositionCommand2["Current"] = 1] = "Current";
  ReplayPositionCommand2[ReplayPositionCommand2["End"] = 2] = "End";
  ReplayPositionCommand2[ReplayPositionCommand2["UnusedPlaceholder"] = 3] = "UnusedPlaceholder";
  return ReplayPositionCommand2;
})(ReplayPositionCommand || {});
var FFBCommand = /* @__PURE__ */ ((FFBCommand2) => {
  FFBCommand2[FFBCommand2["MaxForce"] = 0] = "MaxForce";
  FFBCommand2[FFBCommand2["UnusedPlaceholder"] = 1] = "UnusedPlaceholder";
  return FFBCommand2;
})(FFBCommand || {});
var CameraFocusCommand = /* @__PURE__ */ ((CameraFocusCommand2) => {
  CameraFocusCommand2[CameraFocusCommand2["FocusAtIncident"] = -3] = "FocusAtIncident";
  CameraFocusCommand2[CameraFocusCommand2["FocusAtLeader"] = -2] = "FocusAtLeader";
  CameraFocusCommand2[CameraFocusCommand2["FocusAtExiting"] = -1] = "FocusAtExiting";
  CameraFocusCommand2[CameraFocusCommand2["FocusAtDriver"] = 0] = "FocusAtDriver";
  return CameraFocusCommand2;
})(CameraFocusCommand || {});
var VideoCaptureCommand = /* @__PURE__ */ ((VideoCaptureCommand2) => {
  VideoCaptureCommand2[VideoCaptureCommand2["TriggerScreenShot"] = 0] = "TriggerScreenShot";
  VideoCaptureCommand2[VideoCaptureCommand2["StartVideoCapture"] = 1] = "StartVideoCapture";
  VideoCaptureCommand2[VideoCaptureCommand2["EndVideoCapture"] = 2] = "EndVideoCapture";
  VideoCaptureCommand2[VideoCaptureCommand2["ToggleVideoCapture"] = 3] = "ToggleVideoCapture";
  VideoCaptureCommand2[VideoCaptureCommand2["ShowVideoTimer"] = 4] = "ShowVideoTimer";
  VideoCaptureCommand2[VideoCaptureCommand2["HideVideoTimer"] = 5] = "HideVideoTimer";
  return VideoCaptureCommand2;
})(VideoCaptureCommand || {});
export {
  BroadcastMessageNames,
  BroadcastMessages,
  CameraFocusCommand,
  CameraState,
  CarLeftRight,
  ChatCommand,
  EngineWarnings,
  FFBCommand,
  GlobalFlags,
  IncidentFlagMask,
  IncidentFlags,
  PaceFlags,
  PaceMode,
  PitCommand,
  PitSvFlags,
  PitSvStatus,
  ReloadTexturesCommand,
  ReplayPositionCommand,
  ReplaySearchCommand,
  ReplayStateCommand,
  SessionState,
  StatusField,
  TelemetryCommand,
  TrackLocation,
  TrackSurface,
  TrackWetness,
  UNLIMITED_LAPS_COUNT,
  UNLIMITED_TIME_VALUE,
  VarTypeBytes,
  VarTypes,
  VarTypesReadable,
  VideoCaptureCommand
};
