// irsdk-node/dist/cjs/index.cjs
"use strict";
var __create = Object.create;
var __defProp = Object.defineProperty;
var __getOwnPropDesc = Object.getOwnPropertyDescriptor;
var __getOwnPropNames = Object.getOwnPropertyNames;
var __getProtoOf = Object.getPrototypeOf;
var __hasOwnProp = Object.prototype.hasOwnProperty;
var __export = (target, all) => {
  for (var name in all)
    __defProp(target, name, { get: all[name], enumerable: true });
};
var __copyProps = (to, from, except, desc) => {
  if (from && typeof from === "object" || typeof from === "function") {
    for (let key of __getOwnPropNames(from))
      if (!__hasOwnProp.call(to, key) && key !== except)
        __defProp(to, key, { get: () => from[key], enumerable: !(desc = __getOwnPropDesc(from, key)) || desc.enumerable });
  }
  return to;
};
var __reExport = (target, mod, secondTarget) => (__copyProps(target, mod, "default"), secondTarget && __copyProps(secondTarget, mod, "default"));
var __toESM = (mod, isNodeMode, target) => (target = mod != null ? __create(__getProtoOf(mod)) : {}, __copyProps(
  // If the importer is in node compatibility mode or this is not an ESM
  // file that has been converted to a CommonJS file using a Babel-
  // compatible transform (i.e. "__esModule" has not been set), then set
  // "default" to the CommonJS "module.exports" for node compatibility.
  isNodeMode || !mod || !mod.__esModule ? __defProp(target, "default", { value: mod, enumerable: true }) : target,
  mod
));
var __toCommonJS = (mod) => __copyProps(__defProp({}, "__esModule", { value: true }), mod);

// src/index.ts
var index_exports = {};
__export(index_exports, {
  IRacingSDK: () => IRacingSDK,
  LogLevel: () => import_native2.LogLevel
});
module.exports = __toCommonJS(index_exports);

// src/irsdk-node.ts
var import_node_console = require("node:console");
var import_native = require("@irsdk-node/native");
var import_types = require("@irsdk-node/types");
var import_js_yaml = require("js-yaml");

// src/utils/sim-status.ts
var import_node_http = __toESM(require("node:http"), 1);
var SIM_STATUS_URI = "http://127.0.0.1:32034/get_sim_status?object=simStatus";
var getSimStatus = () => new Promise((resolve, reject) => {
  const req = import_node_http.default.get(SIM_STATUS_URI, (res) => {
    let data = "";
    res.on("data", (d) => {
      data += d;
    });
    res.on("end", () => {
      if (typeof data !== "string") {
        reject(new Error("Invalid payload from sim received"));
      }
      resolve(data.includes("running:1"));
    });
  });
  req.on("error", (err) => {
    reject(err);
  });
});

// src/irsdk-node.ts
function copyTelemData(src, key, dest) {
  dest[key] = { ...src };
  if (src.varType === 1) {
    dest[key].value = [];
    const arr = new Int8Array(src.value);
    arr.forEach((val, i) => {
      dest[key].value[i] = !!val;
    });
    return;
  }
  if (src.varType === 2 || src.varType === 3) {
    dest[key].value = [...new Int32Array(src.value)];
  } else if (src.varType === 4) {
    dest[key].value = [...new Float32Array(src.value)];
  } else if (src.varType === 5) {
    dest[key].value = [...new Float64Array(src.value)];
  }
}
var DefaultConfig = {
  logLevel: import_native.LogLevel.None,
  autoEnableTelemetry: false,
  useTelemVariableCache: true
};
var IRacingSDK = class _IRacingSDK {
  // Public
  /**
   * Enable attempting to auto-start telemetry when starting the SDK (if it is not running).
   * @default false
   */
  autoEnableTelemetry = DefaultConfig.autoEnableTelemetry;
  /**
   * The logging level to use when calling irsdk-node API's. Defaults to 0 (LogLevel.None).
   * @default 0
   */
  logLevel = DefaultConfig.logLevel;
  /**
   * Whether or not to use an internal look-up cache when fetching Telemetry Variables by
   * name. This can provide a performance benefit, but may produce unwanted behaviour when
   * enabled in long-running processes where access of niche, car-specific variables over
   * multiple sessions is common.
   *
   * When enabled, if being used in long-running processes it is recommended to clear the
   * cache whenever you detect the player has changed cars.
   *
   * @default true
   */
  useTelemVariableCache = DefaultConfig.useTelemVariableCache;
  // Private
  _dataVer = -1;
  _sessionData = null;
  _sdk;
  _resolvedConfig;
  _variableIndexCache = {};
  constructor(config) {
    this._resolvedConfig = {
      ...DefaultConfig,
      ...config ?? {}
    };
    const loggingLevel = this._resolvedConfig.logLevel ?? DefaultConfig.logLevel;
    const autoEnableTelemetry = this._resolvedConfig.autoEnableTelemetry ?? DefaultConfig.autoEnableTelemetry;
    const useTelemVariableCache = this._resolvedConfig.useTelemVariableCache ?? DefaultConfig.useTelemVariableCache;
    this._sdk = new import_native.NativeSDK();
    this._sdk.logLevel = loggingLevel;
    this.autoEnableTelemetry = autoEnableTelemetry;
    this.useTelemVariableCache = useTelemVariableCache;
    void _IRacingSDK.IsSimRunning();
  }
  /**
   * Gets the cached variable index from the internal cache, if it exists, otherwise
   * requests the index from the native module.
   *
   * @param varName The variable to grab from the cache.
   * @returns The index of the variable in the variable list.
   */
  _fetchVariableIndexFromCache(varName) {
    const cachedIndex = this._variableIndexCache[varName];
    if (typeof cachedIndex === "number") {
      return cachedIndex;
    }
    const currentIndex = this._sdk.getTelemetryVariableIndex(varName);
    if (currentIndex === null) {
      return null;
    }
    this._variableIndexCache[varName] = currentIndex;
    return currentIndex;
  }
  /**
   * Wait for the SDK module to resolve and load.
   * @deprecated This is no longer needed as of v4.0.3. Please remove.
   */
  async ready() {
    return Promise.resolve(true);
  }
  /**
   * The current version number of the session data. Increments internally every time data changes.
   * @property {number}
   * @readonly
   */
  get currDataVersion() {
    return this._sdk.currDataVersion;
  }
  /** Whether or not to enable verbose logging in the SDK.
   * @property {boolean}
   */
  get enableLogging() {
    return this._sdk.logLevel !== import_native.LogLevel.None;
  }
  set enableLogging(value) {
    this._sdk.logLevel = value ? import_native.LogLevel.Error : import_native.LogLevel.None;
  }
  // @todo: add getter for current session string version
  /**
   * Checks whether the simulation service is running.
   * @returns {boolean} True if the service is running.
   */
  static async IsSimRunning() {
    try {
      const result = await getSimStatus();
      return result;
    } catch (e) {
      (0, import_node_console.error)("Could not successfully determine sim status:", e);
    }
    return false;
  }
  get sessionStatusOK() {
    return this._sdk.isRunning();
  }
  /**
   * Starts the native iRacing SDK and begins subscribing for data.
   * @returns {boolean} If the SDK started successfully.
   */
  startSDK() {
    this.resetTelemetryVariableCache();
    if (!this._sdk.isRunning()) {
      const successful = this._sdk.startSDK();
      if (this.autoEnableTelemetry) {
        this.enableTelemetry(true);
      }
      return successful;
    }
    return true;
  }
  /**
   * Stops the SDK from running and resets the data version.
   */
  stopSDK() {
    this._sdk.stopSDK();
    this._dataVer = -1;
    this.resetTelemetryVariableCache();
  }
  /**
   * Wait for new data from the sdk.
   * @param {number} timeout Timeout (in ms). Max is 60fps (1/60)
   */
  waitForData(timeout) {
    const result = this._sdk.waitForData(timeout);
    if (!result && this._sdk.currDataVersion === -1) {
      this._dataVer = -1;
      this._sessionData = null;
    }
    return result;
  }
  /**
   * Gets the current session data (from yaml format).
   * @returns {SessionData}
   */
  getSessionData() {
    if (this._sessionData && this._dataVer === this.currDataVersion)
      return this._sessionData;
    try {
      const seshString = this._sdk.getSessionData();
      this._sessionData = (0, import_js_yaml.load)(seshString.replaceAll(": ,", ": 0,"));
      return this._sessionData;
    } catch (err) {
      (0, import_node_console.error)("There was an error getting session data:", err);
    }
    return null;
  }
  /**
   * Gets the version number of the latest session data from the SDK.
   */
  getSessionVersionNum() {
    return this._sdk.getSessionVersionNum();
  }
  /**
   * Gets the ID for the current (or previous, if none active) connection.
   */
  getSessionConnectionID() {
    return this._sdk.getSessionConnectionID();
  }
  /**
   * Gets the current weekend info from the session data
   * @returns {WeekendInfo}
   */
  getWeekendInfo() {
    const session = this.getSessionData();
    return session?.WeekendInfo ?? null;
  }
  /**
   * Gets the current session info from the session data.
   * @returns {SessionInfo}
   */
  getSessionInfo() {
    const session = this.getSessionData();
    return session?.SessionInfo ?? null;
  }
  /**
   * Gets the current camera info from the session data.
   * @returns {CameraInfo}
   */
  getCameraInfo() {
    const session = this.getSessionData();
    return session?.CameraInfo ?? null;
  }
  /**
   * Gets the current radio info from the session data.
   * @returns {RadioInfo}
   */
  getRadioInfo() {
    const session = this.getSessionData();
    return session?.RadioInfo ?? null;
  }
  /**
   * Gets the current driver info from the session data.
   * @returns {DriverInfo}
   */
  getDriverInfo() {
    const session = this.getSessionData();
    return session?.DriverInfo ?? null;
  }
  /**
   * Gets the current split time info from the session data.
   * @returns {SplitTimeInfo}
   */
  getSplitInfo() {
    const session = this.getSessionData();
    return session?.SplitTimeInfo ?? null;
  }
  /**
   * Gets the current session info from the session data.
   * @returns {CarSetupInfo}
   */
  getCarSetupInfo() {
    const session = this.getSessionData();
    return session?.CarSetup ?? null;
  }
  /**
   * Get the current value of the telemetry variables.
   *
   * Telemetry gets updated every tick. This is a large object, so large amounts
   * of processing between ticks should attempt to cache this data instead of
   * re-requesting it via this function.
   */
  getTelemetry() {
    const rawData = this._sdk.getTelemetryData();
    const data = {};
    if (Object.keys(rawData).length > 0) {
      Object.keys(rawData).forEach((dataKey) => {
        copyTelemData(
          rawData[dataKey],
          dataKey,
          data
        );
      });
    }
    return data;
  }
  getTelemetryVariable(telemVar) {
    let resolvedTelemVar = telemVar;
    if (this.useTelemVariableCache && typeof telemVar === "string") {
      const cachedIndex = this._fetchVariableIndexFromCache(telemVar);
      if (cachedIndex === null) {
        return null;
      }
      resolvedTelemVar = cachedIndex;
    }
    const rawData = this._sdk.getTelemetryVariable(resolvedTelemVar);
    if (!rawData) {
      return null;
    }
    const parsed = {};
    copyTelemData(
      rawData,
      // eslint-disable-line
      rawData.name,
      parsed
    );
    return parsed[rawData.name];
  }
  /**
   * Resets the internal telemetry variable lookup cache. This occurs automatically
   * whenever the SDK starts and stops, and is only necessary to call if:
   *
   * - `.useTelemVariableCache` is enabled via the `Config` passed to the `IRacingSDK` constructor
   * or by the propertyy on the IRacingSDK instance.
   * - The SDK is being used in a long-running process where potentially niche variables (variables
   * only available for one car) are frequently accessed and the player changes between cars with
   * these niche variables frequently.
   *
   * If that is the case, this function should be called whenever it is detected that the player
   * has changed cars, to make sure there are no stale variables in the cache.
   */
  resetTelemetryVariableCache() {
    this._variableIndexCache = {};
  }
  // Broadcast commands
  enableTelemetry(enabled) {
    const command = enabled ? import_types.TelemetryCommand.Start : import_types.TelemetryCommand.Stop;
    this._sdk.broadcast(import_types.BroadcastMessages.TelemCommand, command);
  }
  restartTelemetry() {
    this._sdk.broadcast(import_types.BroadcastMessages.TelemCommand, import_types.TelemetryCommand.Restart);
  }
  changeCameraPosition(position, group, camera) {
    this._sdk.broadcast(import_types.BroadcastMessages.CameraSwitchPos, position, group, camera);
  }
  // @todo: needs to be padded
  changeCameraNumber(driver, group, camera) {
    this._sdk.broadcast(import_types.BroadcastMessages.CameraSwitchNum, driver, group, camera);
  }
  changeCameraState(state) {
    this._sdk.broadcast(import_types.BroadcastMessages.CameraSetState, state);
  }
  changeReplaySpeed(speed, slowMotion) {
    this._sdk.broadcast(import_types.BroadcastMessages.ReplaySetPlaySpeed, speed, slowMotion ? 1 : 0);
  }
  changeReplayPosition(position, frame) {
    this._sdk.broadcast(import_types.BroadcastMessages.ReplaySetPlayPosition, position, frame);
  }
  searchReplay(command) {
    this._sdk.broadcast(import_types.BroadcastMessages.ReplaySearch, command);
  }
  changeReplayState(state) {
    this._sdk.broadcast(import_types.BroadcastMessages.ReplaySetState, state);
  }
  triggerReplaySessionSearch(session, time) {
    this._sdk.broadcast(import_types.BroadcastMessages.ReplaySearchSessionTime, session, time);
  }
  reloadAllTextures() {
    this._sdk.broadcast(import_types.BroadcastMessages.ReloadTextures, import_types.ReloadTexturesCommand.All, 0);
  }
  reloadCarTextures(car) {
    this._sdk.broadcast(
      import_types.BroadcastMessages.ReloadTextures,
      import_types.ReloadTexturesCommand.CarIndex,
      car
    );
  }
  triggerChatState(state) {
    this._sdk.broadcast(import_types.BroadcastMessages.ChatCommand, state);
  }
  /**
   * @param {number} macro Between 1 - 15
   */
  triggerChatMacro(macro) {
    const clamped = Math.min(15, Math.max(1, macro));
    this._sdk.broadcast(import_types.BroadcastMessages.ChatCommand, import_types.ChatCommand.Macro, clamped);
  }
  triggerPitClearCommand(command) {
    this._sdk.broadcast(import_types.BroadcastMessages.PitCommand, command, 0);
  }
  triggerPitCommand(command) {
    this._sdk.broadcast(import_types.BroadcastMessages.PitCommand, command, 0);
  }
  triggerPitChange(command, amount) {
    this._sdk.broadcast(import_types.BroadcastMessages.PitCommand, command, amount);
  }
  changeFFB(mode, amount) {
    this._sdk.broadcast(import_types.BroadcastMessages.FFBCommand, mode, amount);
  }
  triggerVideoCapture(command) {
    this._sdk.broadcast(import_types.BroadcastMessages.VideoCapture, command);
  }
  /**
   * Trigger a broadcast manually, without any safety guard rails. Only use if
   * you know what you are doing!
   *
   * The function still uses the SDK type map for type-awareness. If you need to
   * turn these off for some reason, toss @ts-expect-error in a command on the
   * line before it to disable the type safety.
   *
   * @param {BroadcastMessages} message The Broadcast Message type.
   * @param args Args for the message. Can be up to 3 numbers.
   */
  broadcastUnsafe(message, ...args) {
    return this._sdk.broadcast(message, ...args);
  }
};

// src/index.ts
__reExport(index_exports, require("@irsdk-node/types"), module.exports);
var import_native2 = require("@irsdk-node/native");
// Annotate the CommonJS export names for ESM import in node:
0 && (module.exports = {
  IRacingSDK,
  LogLevel,
  ...require("@irsdk-node/types")
});
